File: backend/internal/domain/service/conciliacao_calc_test.go
package service

import "testing"

func TestCalcConciliacaoTotais_Quitada(t *testing.T) {
	totais := CalcConciliacaoTotais(10000, 8000, 10000)
	if !totais.Quitada {
		t.Fatal("expected quitada")
	}
	if totais.SaldoEmAberto != 0 {
		t.Fatalf("saldo=%v", totais.SaldoEmAberto)
	}
}

func TestCalcConciliacaoTotais_SaldoEmAberto(t *testing.T) {
	totais := CalcConciliacaoTotais(10000, 5000, 3000)
	if totais.SaldoEmAberto != 7000 {
		t.Fatalf("saldo=%v want 7000", totais.SaldoEmAberto)
	}
	if totais.Quitada {
		t.Fatal("should not be quitada")
	}
}

func TestDeriveNotaFiscalStatus(t *testing.T) {
	if DeriveNotaFiscalStatus(100, 100) != "Pago" {
		t.Fatal("full payment")
	}
	if DeriveNotaFiscalStatus(100, 50) != "Pago Parcial" {
		t.Fatal("partial")
	}
	if DeriveNotaFiscalStatus(100, 0) != "Pendente" {
		t.Fatal("pending")
	}
}

File: src/lib/conciliacao/conciliacaoCalc.test.ts
import { describe, it, expect } from 'vitest';
import { calcConciliacaoTotais, deriveNotaFiscalStatus } from './conciliacaoCalc';

describe('conciliacaoCalc', () => {
  it('calcula saldo em aberto e quitada', () => {
    const t = calcConciliacaoTotais(10000, 8000, 10000, 3);
    expect(t.quitada).toBe(true);
    expect(t.saldoEmAberto).toBe(0);
    expect(t.qtdNotas).toBe(3);
  });

  it('saldo em aberto quando pagamento parcial da ação', () => {
    const t = calcConciliacaoTotais(10000, 5000, 3000);
    expect(t.saldoEmAberto).toBe(7000);
    expect(t.quitada).toBe(false);
  });

  it('deriveNotaFiscalStatus', () => {
    expect(deriveNotaFiscalStatus(100, 100)).toBe('Pago');
    expect(deriveNotaFiscalStatus(100, 40)).toBe('Pago Parcial');
    expect(deriveNotaFiscalStatus(100, 0)).toBe('Pendente');
  });
});
